basics/didResolve.ts

```diff
@@ -0,0 +1,25 @@
+import { DidResolver } from '@kiltprotocol/sdk-js'
+import { Did, DidDocument } from '@kiltprotocol/types'
+
+export async function didResolve(did: Did): Promise<DidDocument | null> {
+  const { didResolutionMetadata, didDocument, didDocumentMetadata } =
+    await DidResolver.resolve(did)
+  if (didDocumentMetadata.deactivated) {
+    console.log(`DID ${did} Document is deactivated`)
+    return null
+  }
+  if (didResolutionMetadata.error) {
+    console.log('Error resolving DID:', didResolutionMetadata.error)
+    return null
+  }
+  if (!didDocument) {
+    console.log('No DID Document found')
+    return null
+  }
```

Comment:
Is this example or test code? When there is no error, there should _always_ be a didDocument, otherwise this is a violation of the contract. So in test code this should raise an error, and in example code this should not be presented as a possibility at all.

Reply:
Ok, as per discussed this now all throws errors.